File: Back-end/src/main/java/br/jus/stj/siscovi/model/TotalMensal.java
package br.jus.stj.siscovi.model;

public class TotalMensal {

    private String funcao;
    private float ferias;
    private float tercoConstitucional;
    private float decimoTerceiro;
    private float incidencia;
    private float multaFGTS;
    private float total;

    public TotalMensal (float ferias, float tercoConstitucional, float decimoTerceiro, float incidencia, float multaFGTS, float total, String funcao) {

        this.ferias = ferias;
        this.tercoConstitucional = tercoConstitucional;
        this.decimoTerceiro = decimoTerceiro;
        this.incidencia = incidencia;
        this.multaFGTS = multaFGTS;
        this.total = total;
        this.funcao = funcao;

    }

    public String getFuncao() {
        return funcao;
    }

    public float getFerias() {
        return ferias;
    }

    public float getTercoConstitucional() {
        return tercoConstitucional;
    }

    public float getDecimoTerceiro() {
        return decimoTerceiro;
    }

    public float getIncidencia() {
        return incidencia;
    }

    public float getMultaFGTS() {
        return multaFGTS;
    }

    public float getTotal() {
        return total;
    }

}

File: Front-end/src/app/totalMensal/totalMensal.ts
export class TotalMensal {
    private _funcao: string;
    private _ferias: number;
    private _tercoConstitucional: number;
    private _decimoTerceiro: number;
    private _incidencia: number;
    private _multaFGTS: number;
    private _total: number;
    private _status: string;
    private _numeroTerceirizados;
    constructor(funcao: string, ferias: number, tercoConstitucional: number, decimoTerceiro: number, incidencia: number, multaFGTS: number, total: number, status: string, numeroTerceirizados: number) {
        this._funcao = funcao;
        this._ferias = ferias;
        this._tercoConstitucional = tercoConstitucional;
        this._decimoTerceiro = decimoTerceiro;
        this._incidencia = incidencia;
        this._multaFGTS = multaFGTS;
        this._total = total;
        this._status = status;
        this._numeroTerceirizados = numeroTerceirizados;
    }


    get funcao(): string {
        return this._funcao;
    }

    get ferias(): number {
        return this._ferias;
    }

    get tercoConstitucional(): number {
        return this._tercoConstitucional;
    }

    get decimoTerceiro(): number {
        return this._decimoTerceiro;
    }

    get incidencia(): number {
        return this._incidencia;
    }

    get multaFGTS(): number {
        return this._multaFGTS;
    }

    get total(): number {
        return this._total;
    }
    get status(): string {
        return this._status;
    }

    get numeroTerceirizados() {
        return this._numeroTerceirizados;
    }
}
